sdcm/cql_stress_cassandra_stress_thread.py

```diff
@@ -0,0 +1,158 @@
+                         stress_num=stress_num, node_list=node_list,  # pylint: disable=too-many-arguments
+                         round_robin=round_robin, stop_test_on_failure=stop_test_on_failure, params=params,
+                         keyspace_num=keyspace_num, keyspace_name=keyspace_name, profile=profile,
+                         client_encrypt=client_encrypt, compaction_strategy=compaction_strategy)
+
+    def create_stress_cmd(self, cmd_runner, keyspace_idx, loader):  # pylint: disable=too-many-branches
+        stress_cmd = self.stress_cmd
+
+        if "no-warmup" not in stress_cmd:
+            # add no-warmup to stress_cmd if it's not there. See issue #5767
+            stress_cmd = re.sub(
+                r'(cql-stress-cassandra-stress [\w]+)', r'\1 no-warmup', stress_cmd)
+
+        if self.keyspace_name:
+            stress_cmd = stress_cmd.replace(
+                " -schema ", " -schema keyspace={} ".format(self.keyspace_name))
+        elif 'keyspace=' not in stress_cmd:  # if keyspace is defined in the command respect that
+            stress_cmd = stress_cmd.replace(
+                " -schema ", " -schema keyspace=keyspace{} ".format(keyspace_idx))
+
+        if self.compaction_strategy and "compaction(" not in stress_cmd:
+            stress_cmd = stress_cmd.replace(
+                " -schema ", f" -schema 'compaction(strategy={self.compaction_strategy})' ")
+
+        if '-col' in stress_cmd:
+            # In cassandra-stress, '-col n=' parameter defines number of columns called (C0, C1, ..., Cn) respectively.
+            # It accepts a distribution as value, however, only 'FIXED' distribution is accepted for the cql mode.
+            # In cql-stress, we decided to accept a number instead.
+            # To support the c-s syntax in test-cases files, we replace occurrences
+            # of n=FIXED(k) to n=k.
+            stress_cmd = re.sub(r' \'n=[\s]*fixed\(([0-9]+)\)\'', r' n=\1',
+                                stress_cmd, flags=re.IGNORECASE)
+
+        if '-rate' in stress_cmd:
+            # In cassandra-stress either '-rate throttle=' or '-rate fixed=' parameter is accepted.
+            # In cql-stress we decided to change it so 'fixed' is a boolean flag
+            # specifying whether the tool should display coordination-omission fixed latencies.
+            stress_cmd = re.sub(r' fixed=[\s]*([0-9]+/s)',
+                                r' throttle=\1 fixed', stress_cmd)
+
+        if '-pop' in stress_cmd:
+            # '-pop seq=x..y' syntax is not YET supported by cql-stress.
+            # TODO remove when seq=x..y syntax is supported.
+            stress_cmd = re.sub(r' seq=[\s]*([\d]+\.\.[\d]+)',
+                                r" 'dist=SEQ(\1)'", stress_cmd)
+
+        if self.node_list and '-node' not in stress_cmd:
+            stress_cmd += " -node "
+            if self.loader_set.test_config.MULTI_REGION:
```

Comment:
It should consider the SCT config option called `region_aware_loader`.

Reply:
this looks like  a code that should be shared, is looks this was copied from cassandra-stress implementation.
and shouldn't (also there an open PR to extend this logic for rack aware configuration)